graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/SchemaGenerator.kt

```diff
@@ -90,14 +90,15 @@ open class SchemaGenerator(internal val config: SchemaGeneratorConfig) {
      * This function is recursive because while generating the additionalTypes it is possible to create additional types that need to be processed.
      */
     protected open fun generateAdditionalTypes(): Set<GraphQLType> {
-        val currentlyProcessedTypes = this.additionalTypes.toSet()
-        this.additionalTypes.clear()
-        val graphqlTypes = currentlyProcessedTypes.map { generateGraphQLType(this, it) }.toSet()
+        val currentlyProcessedTypes = mutableSetOf<KType>()
+        val graphqlTypes = mutableSetOf<GraphQLType>()
+        do {
```

Comment:
i think you want to use `while` loop instead of `do` as additional types could be empty

Reply:
Whoops, merged to fast. Yes it would execute the first time, it would iterate over an empty set so not doing to much, but a while could be clearer